docs/source/user_guide/solver_configuration.rst

```diff
@@ -0,0 +1,195 @@
+=============================
+Solver configuration tutorial
+=============================
+
+Problem definition
+------------------
+
+This tutorial assumes following sample optimization problem definition:
+
+.. code-block:: yaml
+
+    problem:
+        type: knapsack
+        max_weight: 2
+        items_weights: [1, 1, 1]
+        items_costs: [2, 2, 1]
+
+Which defines `knapsack problem <https://en.wikipedia.org/wiki/Knapsack_problem>`_ of filling a knapsack with three items, each characterized with weight and cost.
+The goal is to put chosen items in the knapsack to achieve maximal cost  with total weight not exceeding  ``max_weight``
+
+
+Solver types
+------------
+
+Basic solver definition requires providing its type. Currently supported types are:
+
+* ``vga`` for variational algorithms
+* ``cqm`` for `D-Wave CQM <https://docs.dwavesys.com/docs/latest/doc_leap_hybrid.html#leap-s-hybrid-solvers>`_
+* ``dqm`` for `D-Wave DQM <https://docs.dwavesys.com/docs/latest/doc_leap_hybrid.html#leap-s-hybrid-solvers>`_
+* ``advantage`` for `D-Wave Advantage <https://docs.dwavesys.com/docs/latest/c_gs_4.html>`_ (currently default advantage_system5.4. is supported)
+
+Sample code for defining type advantage solver
+
+.. code-block:: yaml
+
+    solver:
+        type: advantage
+
+
+Configuring initial QUBO penalties (Lagrangian multipliers)
+-----------------------------------------------------------
+
+``advantage`` solver requires problem definition in the `QUBO <https://arxiv.org/pdf/1811.11538>`_ form. QHyper automatically creates the QUBO for
+knapsack problem (for details see
+`Software Aided Approach for Constrained Optimization Based on QAOA Modifications <https://link.springer.com/chapter/10.1007/978-3-031-36030-5_10>`_. )
+
+This, however, requires setting  three penalties (Lagrangian multipliers) i.e. hyperparameters
+for the cost function and two constraints: ensuring that problem encoding is correct and that knapsack weight fullfils ```max_weight``` requirement .
+
+In the example below, the constraint penalties  are set as ``hyper_args``
+
+.. code-block:: yaml
+
+    solver:
+        type: advantage
+        params_inits:
+            hyper_args: [1, 2.5, 2.5]
+
+Adding hyperoptimiser
+---------------------
+
+Since guessing correct penalties is often difficult, there is also option to define ``hyper_optimiser`` to search for appropriate settings.
```

Comment:
```suggestion
Since guessing correct penalties is often difficult, there is also option to define ``hyper_optimizer`` to search for appropriate settings.
```

Reply:
Tak, ma być "z" :-)